ntoskrnl/ps/quota.c

```diff
@@ -143,45 +526,279 @@ PspInsertQuotaBlock(
     KeReleaseSpinLock(&PspQuotaLock, OldIrql);
 }
 
+/**
+ * @brief
+ * Dereferences the quota block that
+ * the process was holding up. This
+ * function is invoked during the
+ * time the process is about to be
+ * destroyed.
+ *
+ * @param[in] Process
+ * The process which is about to be
+ * destroyed.
+ *
+ * @return
+ * Nothing.
+ */
+VOID
+NTAPI
+PspDereferenceProcessQuotaBlock(
+    _In_ PEPROCESS Process)
+{
+    ULONG PsQuotaTypeIndex;
+
+    /* Iterate over the process quota types */
```

Comment:
This seems to be just a debug-only sanity-check loop. (Note that the assert wouldn't do anything in release mode, and the for-loop would be therefore useless.)